doc/testing_snitch.md

```diff
@@ -0,0 +1,146 @@
+# Testing _snitch_
+
+<!-- MarkdownTOC autolink="true" -->
+
+- [Why](#why)
+- [The _snitch_ test suite](#the-snitch-test-suite)
+    - [Run-time unit tests](#run-time-unit-tests)
+    - [Compile-time unit tests](#compile-time-unit-tests)
+    - [Approval tests](#approval-tests)
+        - [How to update approval test data](#how-to-update-approval-test-data)
+    - [Install tests](#install-tests)
+- [How to run the tests](#how-to-run-the-tests)
+    - [Running tests locally](#running-tests-locally)
+        - [Run-time unit tests](#run-time-unit-tests-1)
+        - [Compile-time unit tests](#compile-time-unit-tests-1)
+        - [Approval tests](#approval-tests-1)
+        - [Install tests](#install-tests-1)
+    - [Automated testing pipeline](#automated-testing-pipeline)
+
+<!-- /MarkdownTOC -->
+
+## Why
+
+If you are trying to [contribute](../CONTRIBUTING.md) to _snitch_, you should run the _snitch_ test suite to ensure your changes have not accidentally broken any existing feature. You should also ensure that any new feature you add is properly tested, so others don't break it with their changes later on.
+
+
+## The _snitch_ test suite
+
+_snitch_ has its own test suite, which contains numerous tests. This test suite is composed of:
+ - [run-time unit tests](#run-time-unit-tests) (in `tests/runtime_tests`)
+ - [compile-time unit tests](#compile-time-unit-tests) (not implemented yet)
+ - [approval tests](#approval-tests) (in `tests/approval_tests`)
+ - [install tests](#install-tests) (in `test/install_tests`)
+
+
+### Run-time unit tests
+
+The purpose of these tests is to check the detailed run-time behavior of _snitch_. This includes testing all foreseeable edge cases, happy path and sad path, etc. These tests contribute to the test coverage analysis.
+
+The run-time unit tests are roughly organized by theme into separate files in the `tests/runtime_tests` directory. New files can be added there if necessary. There is no enforced mapping between test files and library source files. Likewise, there is no enforced mapping between test case names and tested classes / functions; this is left of the best judgment of the contributor.
+
+Run-time unit tests come in two flavor: using _doctest_ to test _snitch_, or using _snitch_ to test itself. The latter is less safe, since a bug in _snitch_ might accidentally make these tests pass (e.g., if the test cases are not running at all), but it usually provides nicer tests results on failure.
```

Comment:
```suggestion
Run-time unit tests come in two flavors: using _doctest_ to test _snitch_, or using _snitch_ to test itself. The latter is less safe, since a bug in _snitch_ might accidentally make these tests pass (e.g., if the test cases are not running at all), but it usually provides nicer test results on failure.
```